tests/returntypes/test_jax_qnode_new.py

```diff
@@ -0,0 +1,2209 @@
+            assert np.allclose(hess, expected_hess, atol=10e-2, rtol=0)
+        else:
+            assert np.allclose(hess, expected_hess, atol=tol, rtol=0)
+
+    def test_hessian_vector_valued_postprocessing(
+        self, dev_name, diff_method, interface, mode, tol
+    ):
+        """Test hessian calculation of a vector valued QNode with post-processing"""
+        if diff_method == "adjoint":
+            pytest.skip("Adjoint does not support second derivative.")
+
+        dev = qml.device(dev_name, wires=1)
+
+        @qnode(dev, diff_method=diff_method, interface="jax", mode=mode, max_diff=2)
+        def circuit(x):
+            qml.RX(x[0], wires=0)
+            qml.RY(x[1], wires=0)
+            return qml.expval(qml.PauliZ(0)), qml.expval(qml.PauliZ(0))
+
+        def cost_fn(x):
+            return x @ jax.numpy.array(circuit(x))
+
+        x = jnp.array(
+            [0.76, -0.87],
+        )
+        res = cost_fn(x)
+
+        a, b = x
+
+        expected_res = x @ jnp.array([np.cos(a) * np.cos(b), np.cos(a) * np.cos(b)])
+        assert np.allclose(res, expected_res, atol=tol, rtol=0)
+
+        grad_fn = jax.grad(cost_fn)
+        g = grad_fn(x)
+
+        expected_g = [
+            np.cos(b) * (np.cos(a) - (a + b) * np.sin(a)),
+            np.cos(a) * (np.cos(b) - (a + b) * np.sin(b)),
+        ]
+        assert np.allclose(g, expected_g, atol=tol, rtol=0)
+        hess = jax.jacobian(grad_fn)(x)
+
+        expected_hess = [
+            [
+                -(np.cos(b) * ((a + b) * np.cos(a) + 2 * np.sin(a))),
+                -(np.cos(b) * np.sin(a)) + (-np.cos(a) + (a + b) * np.sin(a)) * np.sin(b),
+            ],
+            [
+                -(np.cos(b) * np.sin(a)) + (-np.cos(a) + (a + b) * np.sin(a)) * np.sin(b),
+                -(np.cos(a) * ((a + b) * np.cos(b) + 2 * np.sin(b))),
+            ],
+        ]
+
+        if diff_method == "finite-diff":
+            assert np.allclose(hess, expected_hess, atol=10e-2, rtol=0)
+        else:
+            assert np.allclose(hess, expected_hess, atol=tol, rtol=0)
+
+    def test_hessian_vector_valued_separate_args(
+        self, dev_name, diff_method, mode, interface, mocker, tol
+    ):
+        """Test hessian calculation of a vector valued QNode that has separate input arguments"""
+        if diff_method == "adjoint":
+            pytest.skip("Adjoint does not support second derivative.")
+
+        dev = qml.device(dev_name, wires=1)
+
+        @qnode(dev, diff_method=diff_method, interface=interface, mode=mode, max_diff=2)
+        def circuit(a, b):
+            qml.RY(a, wires=0)
+            qml.RX(b, wires=0)
+            return qml.probs(wires=0)
+
+        a = jnp.array(1.0)
+        b = jnp.array(2.0)
+        res = circuit(a, b)
+
+        expected_res = [0.5 + 0.5 * np.cos(a) * np.cos(b), 0.5 - 0.5 * np.cos(a) * np.cos(b)]
+        assert np.allclose(res, expected_res, atol=tol, rtol=0)
+
+        jac_fn = jax.jacobian(circuit, argnums=[0, 1])
```

Comment:
Could we have a test case where `argnums` is a subset of all trainable parameters? E.g., having 3 trainable params, but doing `agnums[0,1]` or so.

That should be the case where we're not efficient because we compute the entire Jacobian.

Reply:
Yes I will add that!